### ISSUE TITLE
Swift 5 / Xcode 10.2 support 

### ISSUE THREAD

**Issue by jklausa** [zendesk/zendesk_sdk_ios#480]
Heyo, I'm gonna be _that_ person ;P

Xcode 10.2 and Swift 5 betas came out today. We'd like to start working on migrating our code ASAP, but we can't build our app because you're shipping a binary Swift framework. `swiftc` doesn't support importing modules built with earlier version of compiler, which leads to build faliures like this:

```
Module compiled with Swift 4.2 cannot be imported by the Swift 5.0 compiler: 

/Users/klausa/Developer/a8c/WordPress-iOS/Pods/ZendeskSDK/ZendeskSDK/4.2/ZendeskSDK.framework/Modules/ZendeskSDK.swiftmodule/x86_64.swiftmodule
```

I understand you won't be able to ship a "released" versions of ZendeskSDK using beta Xcode builds.  It would be great if you could provide versions of your "current" SDK built with the new compiler, so we would be unblocked from working on our own migrations.

Thanks in advance!

**Comment by RonanMcH** [comment]
Hey guys, we are working on getting a 10.2 beta release of the SupportSDK for you guys.  Keep an eye on this space. 
Thanks,
Ronan.

**Comment by michaelfransen** [comment]
Any updates?

**Comment by codedbypm** [comment]
+1

**Comment by michaelfransen** [comment]
Hey, can you please provide a timeline for this? 

Thanks,
Michael

**Comment by earthbound19** [comment]
+1

Lack of updated library will halt production if this is not ready when the next (incremental) release of iOS is out (in beta Xcode and iOS it will not compile without a Swift version upgrade; it coerces upgrade). If that becomes costly (goes on for days) we would be forced to jettison ZenDesk.

**Comment by moger777** [comment]
Is it possible to access zendesk via an API vs an SDK? Would make it easier to keep internal code up to date without having to wait for zendesk to release. Obviously down side is a bit of client side code would have to be built up.

**Comment by brendan-fahy** [comment]
Hi all, latest update is still the same, I'm afraid - we haven't got a build for 10.2 yet. I'll re-flag this with the folks internally to see if we can get a better update soon. 
@moger777 we do have a Rest API you can use if you prefer: https://developer.zendesk.com/rest_api/docs/zendesk-apis/resources Though obviously, as you said, that's a significantly higher investment of effort on your side (depending on your integration).

**Comment by moger777** [comment]
Hey @brendan-fahy, has zendesk considered open sourcing this tool? I know other vendors we work with do that which allows the greater community to help submit pull requests for small things like bug fixes and xcode upgrades. At the very least, it would allow us to recompile when new versions of xcode come out.

**Comment by RonanMcH** [comment]
Hi guys,
We released a build of the SupportSDK for swift 5.0 and xcode 10.2 beta 4.  It may not be stable due to the fact it is a beta. It is on a branch called `cocoapods_support_swift_5`. It has Cocoapods support so point your podfiles at that branch.
We will leave this issue open for visability.

Hope this helps,
Ronan.

**Comment by sinoru** [comment]
FYI, Xcode 10.2 is released just minutes ago!

It’ll be great if there are new build which build with stabe Xcode 10.2.

**Comment by michaelfransen** [comment]
Has anyone ran into an issue where this builds if you are building for a device, but not if you build for the simulator? I get this ` cannot find interface declaration for 'ZDKTheme'`.

**Comment by CraigSiemens** [comment]
Using the above branch, I'm always getting the warning `No such module 'ZendeskSDK'` anywhere I use `import ZendeskSDK`. Tried it with a new project and it has the same issue.

This is using Xcode 10.2, Zendesk is added using CocoaPods 1.6.1
```
pod 'ZendeskSDK', :git => 'https://github.com/zendesk/zendesk_sdk_ios.git', :branch => 'cocoapods_support_swift_5'
```

**Comment by CraigSiemens** [comment]
Looks like it should be an easy fix. In the pod spec file it sets the preserved paths

https://github.com/zendesk/zendesk_sdk_ios/blob/8fb30d28da2297b3cd5db5dbba46a91467af83b7/ZendeskSDK.podspec#L33

The issue is that the definition of `s.swift_version`
https://github.com/zendesk/zendesk_sdk_ios/blob/8fb30d28da2297b3cd5db5dbba46a91467af83b7/ZendeskSDK.podspec#L23

doesn't match the folder name `5.0`

https://github.com/zendesk/zendesk_sdk_ios/tree/cocoapods_support_swift_5/ZendeskSDK/5.0/ZendeskSDK.framework

**Comment by pddkhanh** [comment]
Hi. Swift 5 released, we're going to migrate, can you help to work on this, please?

**Comment by idreamer** [comment]
Sure. Let me know how I can help this.

**Comment by idreamer** [comment]
> Hi. Swift 5 released, we're going to migrate, can you help to work on this, please?

Sure. Let me know how I can help this.

**Comment by idreamer** [comment]
> Using the above branch, I'm always getting the warning `No such module 'ZendeskSDK'` anywhere I use `import ZendeskSDK`. Tried it with a new project and it has the same issue.
> 
> This is using Xcode 10.2, Zendesk is added using CocoaPods 1.6.1
> 
> ```
> pod 'ZendeskSDK', :git => 'https://github.com/zendesk/zendesk_sdk_ios.git', :branch => 'cocoapods_support_swift_5'
> ```

I can't download this branch.

**Comment by CameronPorter95** [comment]
> Looks like it should be an easy fix.
> 
> Looking the the files added in the `Pods` folder, I noticed that there are no Zendesk framework files. In the pod spec file it sets the preserved paths
> 
> [zendesk_sdk_ios/ZendeskSDK.podspec](https://github.com/zendesk/zendesk_sdk_ios/blob/8fb30d28da2297b3cd5db5dbba46a91467af83b7/ZendeskSDK.podspec#L33)
> Line 33 in [8fb30d2](/zendesk/zendesk_sdk_ios/commit/8fb30d28da2297b3cd5db5dbba46a91467af83b7)
>  ss.preserve_paths = "ZendeskSDK/#{s.swift_version.to_s}/ZendeskSDK.framework" 
> The issue is that the definition of `s.swift_version`
> 
> [zendesk_sdk_ios/ZendeskSDK.podspec](https://github.com/zendesk/zendesk_sdk_ios/blob/8fb30d28da2297b3cd5db5dbba46a91467af83b7/ZendeskSDK.podspec#L23)
> Line 23 in [8fb30d2](/zendesk/zendesk_sdk_ios/commit/8fb30d28da2297b3cd5db5dbba46a91467af83b7)
>  s.swift_version = "5" 
> doesn't match the folder name `5.0`
> 
> https://github.com/zendesk/zendesk_sdk_ios/tree/cocoapods_support_swift_5/ZendeskSDK/5.0/ZendeskSDK.framework

Confirmed working after this change. I updated my fork (https://github.com/CameronPorter95/zendesk_sdk_ios/tree/cocoapods_support_for_version_swift_5.0) and changed the s.swift_version string to 5.0 and now it pulls down the framework files and compiles with swift 5.0.

Not ideal since it's still a beta but at least it's building for now.

**Comment by bryan1anderson** [comment]
This is pretty frustrating. Seems like Zendesk is always the pod that I have to find tricks to get to even work

**Comment by idreamer** [comment]
I am not sure why installing it with CocoaPod doesn't work in my case. 

```
pod 'ZendeskSDK', :git => 'https://github.com/zendesk/zendesk_sdk_ios.git', :branch => 'cocoapods_support_swift_5'
``` 

So I just installed it manually and now it works.

**Comment by jaz-ah** [comment]
@RonanMcH any timeline on when we can get an official zendesk sdk for xcode10.2 at this point - we're blocked on moving forward on this one

**Comment by idreamer** [comment]
I could make the build success using a manual install. But when I run my app it crashes and shows the error below. 

``` 
dyld: Library not loaded: @rpath/ZendeskCoreSDK.framework/ZendeskCoreSDK
```

**Comment by idreamer** [comment]
> I could make the build success using a manual install. But when I run my app it crashes and shows the error below.
> 
> ```
> dyld: Library not loaded: @rpath/ZendeskCoreSDK.framework/ZendeskCoreSDK
> ```

I added Zendesk Frameworks to **General > Embedded Binaries** manually. Now it works.

**Comment by skydivedan** [comment]
I had success with @CameronPorter95 fork. He found the problem by slightly modifying the `podspec`. Works like a champ.

**Comment by skydivedan** [comment]
also, looks like there was a PR opened against this, which has the change, too.

**Comment by idreamer** [comment]
I am trying to install it using CocoaPod. By the way the install process is stuck on the ```
Pre-downloading: `ZendeskSDK` from `git@github.com:zendesk/zendesk_sdk_ios.git`, branch `cocoapods_support_swift_5`
```

**Comment by nhelke** [comment]
I would very much like to know why Zendesk ignored the fact that Swift did not have a [stable ABI][1] and continues to this day not to have [_module stability_][2] when producing its SDK.

A lot of our developers, who updated to Xcode 10.2, are now unable to build our project as it depends on Zendesk and there is no tagged version built with the release build of Xcode 10.2. In the absence of module stability, I am not prepared to push the Zendesk SDKs build with beta 4 to our master branch, as there is no guarantee that the swiftmodule from beta 4 will be compatible with the release build.

[1]: https://swift.org/blog/abi-stability-and-more/
[2]: https://swift.org/blog/abi-stability-and-more/#module-stability

**Comment by fmessina** [comment]
Basically Zendesk broke the builds of anyone who updated to Xcode 10.2

This should be top priority now, can we have someone from Zendesk give us an update?

**Comment by alexey-ios** [comment]
I've managed to get my application building after manually adding Zendesk from cocoapods_support_swift_5 brabch, but only if I comment out `ZDKZendesk initializeWithAppId` method.
Otherwise I get `No known class method for selector 'initializeWithAppId:clientId:zendeskUrl:' `
Does anyone know how to deal with it in Obj-C AppDelegate? BTW, I've done all the necessary imports:
`@import ZendeskSDK;
@import ZendeskCoreSDK;`
Thank you in advance

**Comment by alena-k** [comment]
Hi everyone,

Providing Swift 5.0 / Xcode 10.2 support is our priority at the moment, and we are working on it.
I can't give you a timeline, unfortunately. But please be assured that we realise the importance of it, and we will provide the new release as soon as we can.

Best regards,
Alena.

**Comment by bryan1anderson** [comment]
> I would very much like to know why Zendesk ignored the fact that Swift did not have a [stable ABI](https://swift.org/blog/abi-stability-and-more/) and continues to this day not to have [_module stability_](https://swift.org/blog/abi-stability-and-more/#module-stability) when producing its SDK.
> 
> A lot of our developers, who updated to Xcode 10.2, are now unable to build our project as it depends on Zendesk and there is no tagged version built with the release build of Xcode 10.2. In the absence of module stability, I am not prepared to push the Zendesk SDKs build with beta 4 to our master branch, as there is no guarantee that the swiftmodule from beta 4 will be compatible with the release build.

At this point it would seem this is more of a capability problem, not a willful choice

**Comment by bryan1anderson** [comment]
> I would very much like to know why Zendesk ignored the fact that Swift did not have a [stable ABI](https://swift.org/blog/abi-stability-and-more/) and continues to this day not to have [_module stability_](https://swift.org/blog/abi-stability-and-more/#module-stability) when producing its SDK.
> 
> A lot of our developers, who updated to Xcode 10.2, are now unable to build our project as it depends on Zendesk and there is no tagged version built with the release build of Xcode 10.2. In the absence of module stability, I am not prepared to push the Zendesk SDKs build with beta 4 to our master branch, as there is no guarantee that the swiftmodule from beta 4 will be compatible with the release build.

@alena-k Plans to include module stability?

**Comment by michaelfransen** [comment]
@alena-k Is there a reason that this wasn't worked on while Xcode 10.2 was beta so it could be ready in time? It is clear that Zendesk is unable to manage their own SDK and should consider making it open source.

**Comment by baz8080** [comment]
Hi all. 

We are in the nearing the end of a sprint here, and the build for Swift 5 will be addressed in the next one. As we've mentioned before, we are aware of the urgency.

@bryan1anderson module stability looks good, and it's likely that we will support this when it is available, and stable. 

@michaelfransen we gave a [build](https://github.com/zendesk/zendesk_sdk_ios/issues/480#issuecomment-472819187) in a previous comment on this issue. The SDK did not build in the first beta of Xcode. Open sourcing of a portion of the SDK is something we're discussing. On another note, I think this language is unhelpful:

> It is clear that Zendesk is unable to manage their own SDK

We are trying to balance supporting you, as well as building new features and SDKs. When someone replies to you here, it's one of the SDK engineers.

**Comment by fmessina** [comment]
@baz8080 thanks for your response.

Apparently Cocoapods support is not working for the branch/build with Swift 5 support. Could you at least fix that, cause most of the Zendesk integrations are probably handled through Cocoapods.

(-> https://github.com/zendesk/zendesk_sdk_ios/pull/494)

To be honest I (as many others I guess) would have expected a solution for this already been released, that is why hearing that it will be fixed in the next sprint (2 weeks?) sounds a bit too long.

**Comment by RonanMcH** [comment]
@fmessina , the branch for the xcode10.2 beta is fixed. Sorry about the delay on this.

**Comment by baz8080** [comment]
@fmessina 

> To be honest I (as many others I guess) would have expected a solution for this already been released

It's a fair point. We did not have it planned in the current sprint, and I'm not sure that Swift 5 being released was a certainty. If it was, then we missed it.

> ...hearing that it will be fixed in the next sprint (2 weeks?) sounds a bit too long.

We would release it as soon as it's done, not wait for the end of the sprint.

**Comment by fmessina** [comment]
@baz8080  @RonanMcH Thanks!
🤗

**Comment by michaelfransen** [comment]
@baz8080 
We do not use cocoapods for Zendesk. I have integrated your beta build into our project and have it working with Xcode 10.2 Do you know of any reason why we shouldn't ship with this?

**Comment by jklausa** [comment]
I appreciate the response @baz8080 and @alena-k — while this isn't the news we'd like to hear, at least it gives us a head's up on what to expect. That's good!

I just wanted to touch one thing:

>We are trying to balance supporting you, as well as building new features and SDKs.

I can't stress this enough — I don't think there's any other "correct" balance here than "drop everything apart from critical security vulnerabilities to make sure our SDKs don't block anyone from updating their toolchains, then resume feature work".  

I understand that you might be under constrains from other parts of the organisation and/or working under some internal deadlines, but, again — I fully believe your **main** responsibility as owners of binary SDKs is to _not break people's builds_. 

It's very worrying that this pattern of delays in supporting new toolchains is repeating.

>We did not have it planned in the current sprint, and I'm not sure that Swift 5 being released was a certainty. If it was, then we missed it.

Swift 5 has had released toolchains since [November 18](https://swift.org/blog/5-0-release-process/) and was considered "done" at that time.(apart from "only select, critical fixes").

Swift 5.1 has been final-branched (again, same caveat about fixes) [a week ago](https://swift.org/blog/5-1-release-process/).

**Comment by RonanMcH** [comment]
Hey @michaelfransen ,
there is no reason you shouldn't release with the beta build other than it was built with beta software released by Apple. It is the same source code as the 2.3.0 Support SDK release. 
Thanks,
Ronan.

**Comment by spaquet** [comment]
Can we have a clear vision of what's going on. What's working (even if it's a dirty fix), what's not working and what's planned?

**Comment by spaquet** [comment]
> @fmessina
> 
> > To be honest I (as many others I guess) would have expected a solution for this already been released
> 
> It's a fair point. We did not have it planned in the current sprint, and I'm not sure that Swift 5 being released was a certainty. If it was, then we missed it.
> 
> > ...hearing that it will be fixed in the next sprint (2 weeks?) sounds a bit too long.
> 
> We would release it as soon as it's done, not wait for the end of the sprint.

Swift 5 was clearly announce month ago. So yes, you missed it and that's a big miss.

**Comment by bryan1anderson** [comment]
This isn’t the first time I’ve been blocked by this SDK. The frustration comes from that, and that you have seemingly admitted your top priority is not to keep your SDK from breaking peoples builds. It wasn’t in this sprint? Why on earth not? 6 betas in Xcode, final branching for Swift 5 cut in November wasn’t enough to signal you guys that you might need to prioritize this in a sprint?

**Comment by harrygt** [comment]
I think moving forward, we would just like a clear indication that Zendesk will not block us from building our apps. We have many pods on our projects yet it's always Zendesk that fails to build when there's a new Swift version. 

As others have said, we appreciate you will likely have pressure from elsewhere in the company but as your customers, we are just voicing our concerns and frustrations with you so you can highlight to you managers, how important it is to not stop our apps from building.

**Comment by alexey-ios** [comment]
> I've managed to get my application building after manually adding Zendesk from cocoapods_support_swift_5 brabch, but only if I comment out `ZDKZendesk initializeWithAppId` method.
> Otherwise I get `No known class method for selector 'initializeWithAppId:clientId:zendeskUrl:' `
> Does anyone know how to deal with it in Obj-C AppDelegate? BTW, I've done all the necessary imports:
> `@import ZendeskSDK; @import ZendeskCoreSDK;`
> Thank you in advance

Sorry, guys, for mentioning again, but I can't figure out, why did this error come up after updating to Swift 5.0 branch version. Did anyone experience such error?

**Comment by welshm** [comment]
Appreciate the info and quick update - not to add fuel to the fire, but we're another company that has now been bitten by Zendesk SDK (more than once) and am happy to provide another voice for you to present to your management to make this a top priority in the future.

**Comment by agordeev** [comment]
We've got 24 pods in our project. Unfortunately, Zendesk SDK is the only one which is forcing me to download back Xcode 10.1 to build the project.

As people saying above, Xcode 10.2 beta was available since January.

**Comment by alexey-ios** [comment]
The same picture here. And now we've got problems with our AppDelegate being written in Obj-C,  which were finally solved by extending it in Swift to initialise Zendesk, as neither Zendesk, nor ZDKZendesk don't work in Obj-C.
I wish I could make decisions on choosing platforms in our company.

**Comment by nhelke** [comment]
@baz8080 @RonanMcH  I have had a thought that could help you provide a stable future-proof version of your framework in the presence of ABI stability but in the absence of module stability. Sadly this won’t help with the Xcode 10.1 to 10.2 transition, but it would provide a framework that would be compatible with all versions of Xcode greater or equal to 10.2.

You could remove the swiftmodule from your framework, as it is a binary unstable format, specific to the version of Xcode that was used to produce it. Instead you could provide only Objective-C bindings. Objective-C headers are a stable format, so the framework would be compatible with Xcode 10.2 and any future versions of Xcode thanks to stable ABI. It should be as simple as including ZendeskSDK-Swift.h in your umbrella header (instead of as a submodule), although you might be able to get nicer bindings by manually editing the header YMMV. 

If you go down this path you should probably add something like the following to ensure older versions of Xcode don’t attempt to use the framework, as calling into ABI-stable libraries from pre-ABI-stable binaries is undefined:

```objc
#if __IPHONE_OS_VERSION_MAX_ALLOWED < 122000
#error This framework is only compatible with Xcode 10.2 and up
#endif
```

**Comment by OpExNetworks** [comment]
I'd ask that Zendesk be proactive on it's builds and factoring in Swift/Xcode updates well into their sprints.  It's not like everyone didn't know that Swift 5 was coming out soon.  It's pretty easy to stay on top of this since, as a developers, we all have access to beta versions of Xcode.  It's not to say that Zendesk sucks, it doesn't but it would be helpful to have builds and the supporting doc ready to go.  Other product companies seem to be able to do this successfully so not sure why Zendesk always seems to be quite behind on releases.

**Comment by postmechanical** [comment]
@alena-k @baz8080 I've got a hybrid Obj-C/Swift project and using `cocoapods_support_swift_5` branch I have same problems as https://github.com/zendesk/zendesk_sdk_ios/issues/480#issuecomment-476750077.

**Comment by chaseacton** [comment]
This has also broken my company's app. It's always one issue after another with the Zendesk SDK. When can we expect an update?

**Comment by kevinmun** [comment]
I have the same problem  as [#480](https://github.com/zendesk/zendesk_sdk_ios/issues/480#issuecomment-476750077) too

**Comment by nekrich** [comment]
Hello.

Xcode 10.2 is released with Swift 5.0. Is there any ETA on release?

**Comment by technikyle** [comment]
This should be ready to go by now. There was a long beta and the developers have had plenty of warning.

**Comment by RonanMcH** [comment]
Hey guys,
We just released version 2.3.1 of the Support SDK with Xcode 10.2 support.
Thanks,
Ronan.

**Comment by postmechanical** [comment]
@RonanMcH Does it resolves the issues described in https://github.com/zendesk/zendesk_sdk_ios/issues/480#issuecomment-476750077?

**Comment by bryan1anderson** [comment]
@RonanMcH Thanks for keeping us updated. Please consider this situation in sprints going forward. It is extremely easy to predict when the release of a new swift version will be, and the community will help do that.

I understand it was most likely a management problem pushing a different agenda and I'm not sure what your internal dev team looks like, but if there are dedicated iOS engineers at all, I believe priority should be to maintain a buildable SDK. 

This seems to happen more often than I'd like. We're approaching 5.1 deployment, and then after that Swift 6. Please make these priorities

**Comment by RonanMcH** [comment]
Hi @postmechanical,
It does fix the issue you are referencing.
Thanks,
Ronan.

**Comment by earthbound19** [comment]
@bryan1anderson for the record, I have no affiliation with ZenDesk or Apple or any other parties directly involved with development of any technology in discussion here.

Your comment is presumptuous and condescending. You know literally nothing about their schedule or management, and "extremely easy" should just never be uttered by any programmer, ever. Programming and project management and everything else involved is _not_ easy by any fair relative metric. This stuff is extraordinarily abstract, abstruse, opaque, advanced, etc--_difficult_. Difficult like you. What a total prick you were in that comment. I mean wow.

**Comment by earthbound19** [comment]
@bryan1anderson that said, pray tell how it is "extremely easy" to predict new Swift (or, it seems to me they may be released in tandem--iOS and Xcode) releases. I have searched high, low, near and far for anything that clearly indicates any release schedule or anticipation of new versions of any of these. Where do you find information on such schedules or expectations?

**Comment by olegklk** [comment]
Having Zendesk installed via Cocoapods in Objective-C project I keep getting an error <ZendeskSDK/ZendeskSDK.h> file not found.
And I see that ZendeskSDK.xcconfig has no definition for `HEADER_SEARCH_PATHS`. Is that expected?
My guess it should be something like `HEADER_SEARCH_PATHS = $(inherited) "${PODS_ROOT}/Headers/Private" "${PODS_ROOT}/Headers/Private/ZendeskSDK" "${PODS_ROOT}/Headers/Public" "${PODS_ROOT}/Headers/Public/ZendeskSDK"`

**Comment by olegklk** [comment]
In fact there is no ZendeskSDK folder inside Pods/Headers/Private or Pods/Headers/Public not sure but seems like all my other pods are presented there

**Comment by RonanMcH** [comment]
@olegklk , checking this out now

**Comment by RonanMcH** [comment]
Hi @olegklk ,
We integrated the Support SDK using Cocoapods (versions 1.5.3 and 1.6.0) with an Objective-c project and they aren't showing any errors.  Is it possible it is an issue with your own set up?
Thanks,
Ronan.

**Comment by m1entus** [comment]
[!] CocoaPods could not find compatible versions for pod "ZendeskSDK":
  In Podfile:
    ZendeskSDK (~> 2.3.1)

**Comment by githubdoramon** [comment]
"Pods update" missing maybe?

**Comment by OpExNetworks** [comment]
This was predictable.  Unable to manually add the frameworks successfully.  Here....
<img width="1123" alt="Screen Shot 2019-04-02 at 2 16 10 PM" src="https://user-images.githubusercontent.com/20602070/55436872-4c762080-5552-11e9-895b-1ca18258d755.png">

**Comment by HamstringAssassin** [comment]
@OpExNetworks, 

That run script is working correctly for us here. Does the directory exist? Can you find the strip-frameworks.sh script there?

**Comment by earthbound19** [comment]
@bryan1anderson your comment is presumptuous and condescending vomit, immediately after ZenDesk delivered. Boo.

I'm not affiliated with anyone making any code around this. I just speak my mind when people are rude.

**Comment by RonanMcH** [comment]
Hey guys, Im going to close this issue. If any issues persist, please open other GitHub issues and we will try our best to help you. 
Thanks,
Ronan.

**Comment by chivalrousbob** [comment]
I still have the same issue @OpExNetworks !

**Comment by alexey-ios** [comment]
@chivalrousbob @OpExNetworks  you should delete this script from your Build Phases if you are now using CocoaPods, but previously have installed Zendesk manually. Just check that you don't have any scripts with Zendesk mentioning there in Build Phases if you're using CocoaPods.
BTW, problem with Obj-C wasn't solved....

**Comment by chivalrousbob** [comment]
thanks @alexey-ios  yes I've forgotten to remove that script when i downloaded it from Cocoapods

**Comment by softmarshmallow** [comment]
> Hey guys,
> We just released version 2.3.1 of the Support SDK with Xcode 10.2 support.
> Thanks,
> Ronan.

> Module compiled with Swift 4.2 cannot be imported by the Swift 5.0 compiler: ~/Pods/ZendeskSDK/ZendeskSDK/5.0/ZendeskSDK.framework/Modules/ZendeskSDK.swiftmodule/arm64.swiftmodule
